### PR TITLE
Esquivwg code review1

### PR DESCRIPTION
**Pull request by Wilmer19**
Analysis of the program.
This is an app that helps with memorizing. 

What I learned from this code review:
I learned how to use Regex to convert strings into a regular expression.
I learned about naming conventions. 
I learned that there should be no comments in the IDE. 
 
Was the program available in UC Github on time?
Yes the program was available in UC Github on time.

Is the program documented/commented well enough for you to understand?
Yes the program is documented well enough to understand.

Does the program compile?
Yes the program does compile.

Rationale behind your changes.
I deleted all comments from the program to reduce technical dept. Comments can be made in the project's GitHub. 
I also deleted an unnecessary unit test to reduce technical dept. It was a practice unit test that we used in class. It didn't do anything for the project.
In ICardDAO, there was a bad use of naming convention. Classes should be UpperCamel case so I changed notecards to noteCards. Here's my resource: https://kotlinlang.org/docs/coding-conventions.html#source-file-organization

My project commits
Adding text: https://github.com/bracko-3/RentMinder/commit/42cde1cf424a77f7cca5b22a5d3b11fafbe9b98d
Adding textfield and button: https://github.com/bracko-3/RentMinder/commit/77f7994ccc7078b6923aa77436c05ad47cc3ca6f
Adding spacer: https://github.com/bracko-3/RentMinder/commit/edb833ab2d9fada2244122a64297abbb3c72d811

### CODE REVIEW

File: app/src/main/java/com/memrepo/dto/NoteCard.kt [hunk 1]
```diff
@@ -4,18 +4,22 @@ import androidx.room.ColumnInfo
...
-
-        //var snippetDisplayList = snippetDisplay.split(" ").toList() create a list by splitting string on space
-        //var snippetDisplayList : List<String> = listOf<String>(this.snippetDisplay.split(" ").toString())
-
-        override fun toString(): String {
-                return snippetDisplay
+        fun createSnippetDisplayList(): List<String> {
+                return this.snippet
+                        .replace("-".toRegex(), " ")
+                        .replace("  ".toRegex(), " ")
```
Comment (Wilmer19) [inline app/src/main/java/com/memrepo/dto/NoteCard.kt:19]: The development branch had the updated code so I copied and pasted it here.

File: app/src/test/java/com/memrepo/ExampleUnitTest.kt [hunk 2]
```diff
@@ -11,49 +11,29 @@ import org.junit.Assert.*
...
-                    "We hold these truths to be self-evident, that all men are created equal.")
-        assertEquals("I have a dream that one day this nation will rise up and live out the true meaning of its creed " +
-                "We hold these truths to be self evident that all men are created equal", iHaveADream.snippetDisplay)
-        assertEquals("nation", iHaveADream.snippetDisplayList[8])
-        assertEquals("creed", iHaveADream.snippetDisplayList[20])
-        assertEquals("truths", iHaveADream.snippetDisplayList[24])
-    }
-
-    @Test
-    fun threePlusThree_EqualsSix()
```
Comment (Wilmer19) [inline app/src/test/java/com/memrepo/ExampleUnitTest.kt:49]: The development branch had the updated code so I copied and pasted it here.

File: app/src/main/java/com/memrepo/dao/ICardDAO.kt [hunk 3]
```diff
@@ -10,11 +10,11 @@ interface ICardDAO {
     fun getAllNoteCards() : List<NoteCard>
 
     @Insert
-    fun saveNoteCard(vararg notecards: NoteCard)
+    fun saveNoteCard(vararg noteCards: NoteCard)
 
     @Update(onConflict = OnConflictStrategy.REPLACE)
-    fun updateNoteCard(vararg notecards: NoteCard)
+    fun updateNoteCard(vararg noteCards: NoteCard)
```
Comment (Wilmer19) [inline app/src/main/java/com/memrepo/dao/ICardDAO.kt:16]: Class names should be UpperCamel case. Here's my resource: https://kotlinlang.org/docs/coding-conventions.html#naming-rules

File: app/src/main/java/com/memrepo/MainActivity.kt [hunk 4]
```diff
@@ -72,13 +68,11 @@ class MainActivity : ComponentActivity() {
                         },
                     )
                 },
-                // Bar across bottom of screen
                 bottomBar = {
                     BottomAppBar {
                         Box(Modifier.fillMaxWidth()) {
                             Button(
                                 onClick = {
-                                    // When the button is clicked the bottom sheet will expand or collapse
```
Comment (Wilmer19) [inline app/src/main/java/com/memrepo/MainActivity.kt:81]: I deleted all comments to reduce technical dept. There shouldn't be any in the IDE.

Comment (discospiff) [reply app/src/main/java/com/memrepo/MainActivity.kt:81]: While too much documentation is considered bad practice, and KDoc is preferred over JavaDoc, these comments aren't necessarily hurting anything.

File: app/src/main/java/com/memrepo/dao/ICardDAO.kt (hunk 3 above)
Comment (discospiff) [reply app/src/main/java/com/memrepo/dao/ICardDAO.kt:16]: Good.

File: app/src/main/java/com/memrepo/dto/NoteCard.kt [hunk 5]
```diff
@@ -4,18 +4,22 @@ import androidx.room.ColumnInfo
 import androidx.room.Entity
 import androidx.room.PrimaryKey
 
+/**
+ * @constructor creates a dto for what values should be stored in a NoteCard.
+ *
+ * cardId is a primary key that autogenerates for every newly created NoteCard.
+ * title and snippet are both components of a NoteCard that is a stored string for the User.
+ */
```
Comment (discospiff) [inline app/src/main/java/com/memrepo/dto/NoteCard.kt:12]: Ok, this is good.  KDoc over public methods/classes is preferred to inline comments.

File: app/src/main/java/com/memrepo/dto/NoteCard.kt (hunk 1 above)
Comment (discospiff) [reply app/src/main/java/com/memrepo/dto/NoteCard.kt:19]: Better to merge branches than copy-paste.  Merging keeps the original context.

File: app/src/test/java/com/memrepo/ExampleUnitTest.kt [hunk 6]
```diff
@@ -11,49 +11,29 @@ import org.junit.Assert.*
...
  */
 class ExampleUnitTest {
-    @Test
-    fun addition_isCorrect() {
-        assertEquals(4, 2 + 2)
-    }
 
     @Test
-    fun toString_NoteCard(){
+    fun `Given a NoteCard about the order of the planets, When I toString the object, the ID, title, Then snippet attributes should be returned`(){
```
Comment (discospiff) [inline app/src/test/java/com/memrepo/ExampleUnitTest.kt:16]: This is better naming, yes.

File: app/src/test/java/com/memrepo/ExampleUnitTest.kt [hunk 7]
```diff
@@ -11,49 +11,29 @@ import org.junit.Assert.*
...
-    {
-        assertEquals(7, 3 + 4)
+    fun `Given a NoteCard with cardID, 0, the title, I Have A Dream Speech, and a snippet, 2 sentences from MLK's speech, When When I use createSnippetDisplayList() to create a list of the snippet words, Then the list should populate each index with its own word, no punctuation, in all lowercase`(){
+        val speech = NoteCard(0, "I Have A Dream Speech",
+            "I have a dream that one day this nation will rise up and live out the true meaning of its creed. " +
+                    "We hold these truths to be self evident that all men are created equal")
+        val speechWordList = speech.createSnippetDisplayList()
+
+        assertEquals("[i, have, a, dream, that, one, day, this, nation, will, rise, up, and, live, out, the, true, meaning, of, its, creed, " +
+                "we, hold, these, truths, to, be, self, evident, that, all, men, are, created, equal]", speechWordList.toString())
```
Comment (discospiff) [inline app/src/test/java/com/memrepo/ExampleUnitTest.kt:37]: This test is brittle, as one changed word invalidates the test.

**Comment by discospiff** [comment]
Good suggestions here, though I'd recommend merging branches over copy-pasting.  See comments in Files Changed.

**Pull request closed without merging**